common/scripts/set-rush-write-cache-variables.js

```diff
@@ -0,0 +1,14 @@
+/**
+ *  Because ADO evaluates Agent.OS at runtime,
+ *  we cannot use it in build-time expressions.
+ *  Instead, set the RUSH_BUILD_CACHE_WRITE_ALLOWED here during runtime.
+ */
+
+const acceptableReasons = ["IndividualCI", "Manual"]
+let useRushWriteCache = 0
+
+if (process.env.AGENT_OS === 'Linux' && acceptableReasons.includes(process.env.BUILD_REASON))
+  useRushWriteCache = 1
+
+console.log(`setting RUSH_BUILD_CACHE_WRITE_ALLOWED to ${useRushWriteCache}`)
+console.log(`##vso[task.setvariable variable=RUSH_BUILD_CACHE_WRITE_ALLOWED;]${useRushWriteCache}`)
```

Comment:
is this supposed to be in a log? where does it actually set the env var?

Reply:
The `##vso...` line sets the variable - it's the equivalent of `echo` in `bash` or `Write-Host` in `powershell`:
https://github.com/iTwin/itwinjs-core/pull/4264

Reply:
TIL!

Reply:
Me too, actually. Thought it was restricted to bash/ps - but this makes sense.